carapace-server/src/main/java/org/carapaceproxy/server/Listeners.java

```diff
@@ -190,12 +209,43 @@ private void bootListener(NetworkListenerConfiguration listener) throws Interrup
                     public void initChannel(SocketChannel channel) throws Exception {
                         CURRENT_CONNECTED_CLIENTS_GAUGE.inc();
                         if (listener.isSsl()) {
-                            SniHandler sni = new SniHandler(sniMappings);
-                            channel.pipeline().addLast(sni);
+                            SniHandler sni = new SniHandler(sniMappings) {
+//                                @Override
+//                                protected Future<SslContext> lookup(ChannelHandlerContext ctx, java.lang.String hostname) throws java.lang.Exception {
+//                                    Future<SslContext> futureSslContext = super.lookup(ctx, hostname);
+//                                    try {
+//                                        SslContext context = (SslContext) futureSslContext.getNow();
+//                                        SslHandler handler = context.newHandler(channel.alloc());
+//                                        ReferenceCountedOpenSslEngine engine = (ReferenceCountedOpenSslEngine) handler.engine();
+//                                        engine.setOcspResponse(ocspMan.getAllOcspResp().get(0).getEncoded());
+//                                        channel.pipeline().addAfter("sni", "ocsp", handler);
+//                                    } catch (IOException ex) {
+//                                        LOG.log(Level.SEVERE, ex + "");
+//                                    }
+//
+//                                    return futureSslContext;
+//                                }
+
+                                @Override
+                                protected SslHandler newSslHandler(SslContext context, ByteBufAllocator allocator) {
+                                    try {
+                                        if (listener.isOcsp() && OpenSsl.isOcspSupported()) {
+                                            SslHandler handler = context.newHandler(allocator);
+                                            ReferenceCountedOpenSslEngine engine = (ReferenceCountedOpenSslEngine) handler.engine();
+                                            engine.setOcspResponse(ocspMan.getAllOcspResp().get(0).getEncoded());
+                                            channel.pipeline().addAfter("sni", "ocsp", handler);
```

Comment:
I can't understand why are you adding a new SslHandler, so you are going to have 2 SslHandler

please add a comment if this is really want we need